eras/shelley/impl/src/Cardano/Ledger/Shelley/TxBody.hs

```diff
@@ -1014,15 +1015,25 @@ instance-- use the weakest constraint necessary
       <> toCBOR addr
       <> toCBOR coin
 
-instance-- use the weakest constraint necessary
-
+-- use the weakest constraint necessary
+instance
   (Era era, TransTxOut DecodeNonNegative era, Show (Core.Value era)) =>
   FromCBOR (TxOut era)
   where
-  fromCBOR = decodeRecordNamed "TxOut" (const 2) $ do
-    cAddr <- fromCBOR
-    coin <- decodeNonNegative
-    pure $ TxOutCompact cAddr coin
+  fromCBOR = fromNotSharedCBOR
+
+-- This instance does not do any sharing and is isomorphic to FromCBOR
```

Comment:
Because of the `FromSharedCBOR (Core.TxOut era)` constraint on this instance:

```
instance
  ( CC.Crypto (Crypto era),
    FromSharedCBOR (Core.TxOut era),
    Share (Core.TxOut era) ~ Interns (Credential 'Staking (Crypto era))
  ) =>
  FromSharedCBOR (UTxO era)
  where
 ....
````

In other words Alonzo `TxOut` (and so will Babbage) has sharing capabilities, while Shelley TxOut is unshareable